### PR TITLE
Fix windows version validator under Chinese

### PR DESCRIPTION
**Pull request by ctrysbita**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

Flutter tool currently uses `OS Version:` to find the installed version of windows from `systeminfo` command, however, the command may return localized string. Take Chinese as an example:
```
主机名:           XXXXXXXXXXXX
OS 名称:          Microsoft Windows 11 专业版
OS 版本:          10.0.22621 暂缺 Build 22621
OS 制造商:        Microsoft Corporation
OS 配置:          独立工作站
OS 构建类型:      Multiprocessor Free
注册的所有人:     暂缺
注册的组织:       暂缺
产品 ID:          XXXXXXXXXXXX
初始安装日期:     2022/11/9, 13:33:50
系统启动时间:     2022/11/30, 13:36:47
系统制造商:       ASUS
系统型号:         System Product Name
系统类型:         x64-based PC
处理器:           安装了 1 个处理器。
                  [01]: Intel64 Family 6 Model 151 Stepping 2 GenuineIntel ~3600 Mhz
BIOS 版本:        American Megatrends Inc. 2103, 2022/9/30
Windows 目录:     C:\WINDOWS
系统目录:         C:\WINDOWS\system32
启动设备:         \Device\HarddiskVolume1
系统区域设置:     zh-cn;中文(中国)
输入法区域设置:   zh-cn;中文(中国)
时区:             (UTC+08:00) 北京，重庆，香港特别行政区，乌鲁木齐
物理内存总量:     65,277 MB
可用的物理内存:   55,333 MB
虚拟内存: 最大值: 75,005 MB
虚拟内存: 可用:   61,781 MB
虚拟内存: 使用中: 13,224 MB
页面文件位置:     C:\pagefile.sys
域:               WORKGROUP
登录服务器:       \\XXXXXXXXXXXX
修补程序:         安装了 3 个修补程序。
                  [01]: KB5020622
                  [02]: KB5019980
                  [03]: KB5019304
Hyper-V 要求:     已检测到虚拟机监控程序。将不显示 Hyper-V 所需的功能。
```
This causes the original approach always fail. The PR updates RegExp to prevent the hardcoded `Version`.

*List which issues are fixed by this PR. You must list at least one issue.*

#116280

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by christopherfujino** [comment]
@eliasyishak please take a look at this one (at a glance seems reasonable)

**Review by eliasyishak — CHANGES_REQUESTED** [review]
Thank you for this PR! I was curious how the results for `systeminfo` would be affected for different parts of the world.

Added a simple fix that can reduce how many groups we use in the regex pattern.

**Review by eliasyishak — APPROVED** [review]
LGTM! Thank you for your contribution!

File: AUTHORS [hunk 1]
```diff
@@ -99,3 +99,4 @@ Nguyễn Phúc Lợi <nploi1998@gmail.com>
 Jingyi Chen <jingyichen@link.cuhk.edu.cn>
 Junhua Lin <1075209054@qq.com>
 Tomasz Gucio <tgucio@gmail.com>
+Jason C.H <ctrysbita@outlook.com>
```
Comment (christopherfujino) [inline AUTHORS:102]: Can you add a trailing newline here?

**Review by christopherfujino — CHANGES_REQUESTED** [review]

File: AUTHORS (hunk 1 above)
Comment (ctrysbita) [reply AUTHORS]: Done! A trailing newline is added.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by CaseyHillers** [comment]
@ctrysbita can you rebase this PR with the latest from tip of tree? The Google testing infra is failing as this PR is 80 commits behind.

**Pull request merged**